engine/src/test/java/io/zeebe/engine/processor/StreamProcessorTest.java

```diff
@@ -544,7 +545,7 @@ public void processRecord(
   }
 
   @Test
-  public void shouldCreateSnapshotEvenWhenNoEventProcessed() throws Exception {
+  public void shouldCreateSnapshotOnCloseEvenWhenNoEventProcessed() throws Exception {
```

Comment:
This tests that, on close, we will create a snapshot even if we did not process anything. This is still true? What the test has updated is that we have to have processed something at some point in history (so position is not -1), but that if this time we didn't process anything between open/close, we'll still create the snapshot. 

Or did I misunderstand what you mean?

Reply:
Isnt that a contradiction ? :sweat_smile: You saying `we will create a snapshot even if we did not process anything.` In the next sentence you say `What the test has updated is that we have to have processed something at some point in history (so position is not -1)` :sweat_smile: 


> but that if this time we didn't process anything between open/close, we'll still create the snapshot.

This is not what the test verifies, then I would expect two open and close cycles at least.

Reply:
What does the second open/close verify that the first doesn't? Let's say we didn't take snapshots on close if we hadn't processed anything since the last snapshot.

1. Process one event
1. Take snapshot
1. Close - no snapshot taken

Now we verify that:

1. Open
1. Process event
1. Take snapshot
1. Close
1. Take snapshot again

Would a second open/close verify something more?

Reply:
Ok I think I misunderstood the test or what you want to verify. 

I think this what you added should be in the `given` section. This is your starting point. It is `given` that you did a snapshot, `when` you close the stream processor `then` you expect one more snapshot even if no processing was in between